python_modules/libraries/dagster-components/dagster_components/core/component_defs_builder.py

```diff
@@ -141,3 +146,19 @@ def load_defs(
         )
         all_defs.append(defs)
     return Definitions.merge(*all_defs)
+
+
+def get_passed_assets_defs(defs: Definitions) -> Sequence[AssetsDefinition]:
+    """Returns all AssetsDefinitions passed to the Definitions object. Does not support
+    CacheableAssetsDefinitions, and ignores bare AssetSpecs.
+    """
+    check.invariant(
+        not any(
+            isinstance(asset_def, CacheableAssetsDefinition) for asset_def in (defs.assets or [])
+        ),
+        "Cannot call get_passed_assets_defs on a Definitions object that contains CacheableAssetsDefinitions",
+    )
+
+    return [
+        asset_def for asset_def in (defs.assets or []) if isinstance(asset_def, AssetsDefinition)
+    ]
```

Comment:
we should canonicalize `AssetSpec` and `SourceAssets` to `AssetsDefinition` here

Reply:
good call, updated